src/anyio/_backends/_asyncio.py

```diff
@@ -126,6 +128,14 @@ def current_task(loop: Optional[asyncio.AbstractEventLoop] = None) -> Optional[a
 
         return asyncio.Task.current_task(loop)
 
+
+# Needed for the workaround for async generator shutdown on Python 3.6
+async def _dummy_asyncgen():
+    yield
+
+async_generator_athrow = _dummy_asyncgen().athrow(Exception).__class__
```

Comment:
Doesn't this throw an unawaited Coroutine warning?

Reply:
Fixed that one.